### PR TITLE
Initial support for using expressions to filter tiledb results

### PR DESCRIPTION
**Pull request by nalinigans**

### CODE REVIEW

File: core/src/expressions/expression.cc [hunk 1]
```diff
@@ -29,103 +30,208 @@
...
+      if (drop_cell(ret)) {
+        cells_to_be_dropped.push_back(i_cell);
+      }
+    } catch (mup::ParserError const &e) {
+      EXPRESSION_ERROR("Parser evaluate error, possibly due to bad filter expression: " + "\n\t" + e.GetMsg());
+      return TILEDB_EXPR_ERR;
     }
   }
 
+  // TODO: support only for sparse arrays intially.
```
Comment (mlathara) [inline core/src/expressions/expression.cc]: Should we warn (or do we already elsewhere) if someone tries this with dense arrays?

File: core/src/expressions/expression.cc [hunk 2]
```diff
@@ -29,103 +30,208 @@
...
+        assert(array_schema_->cell_size(0) == sizeof(int));
+        mup::Value x_int(mup::int_type(0));
+        attribute_map_.insert(std::make_pair(name, x_int));
+        break;
+      }
+      case TILEDB_FLOAT32: {
+        mup::Value x_float(mup::float_type(0.0));
+        attribute_map_.insert(std::make_pair(name, x_float));
+        break;
+      }
```
Comment (mlathara) [inline core/src/expressions/expression.cc]: Should we have a default case here?

Comment (nalinigans) [reply core/src/expressions/expression.cc]: There really is no default case now, as we just ignore the attributes that are not going to be supported in the expressions for now. I can't really print out that the attribute is not going to be supported in the expressions as I will have to parse the expression for that.

File: core/src/expressions/expression.cc (hunk 1 above)
Comment (nalinigans) [reply core/src/expressions/expression.cc]: We are not warning yet, it is silently ignored now. Will add a message here. Thanks.

**Pull request merged**